song-server/src/main/java/bio/overture/song/server/service/analysis/AnalysisServiceSender.java

```diff
@@ -174,6 +169,6 @@ public Analysis securedDeepRead(@NonNull String studyId, String id) {
   private void sendAnalysisMessage(
       Analysis analysis, AnalysisStates analysisState, AnalysisActions action) {
     val message = createAnalysisMessage(action, analysis, songServerId);
-    sender.send(toJson(message));
+    sender.send(toJson(message), message.getAnalysisId());
```

Comment:
Only place in the code base where kafka messages are sent. We specify the key to be the AnalysisID from the message.

Reply:
looks good to me 👍